sql/size_utils.sql

```diff
@@ -126,151 +184,280 @@ BEGIN
+               total_size  TEXT,
+               node_name   NAME) LANGUAGE PLPGSQL STABLE STRICT
+               AS
 $BODY$
 DECLARE
+        table_name       NAME;
+        schema_name      NAME;
 BEGIN
-    IF NOT _timescaledb_internal.dimension_is_finite(time_value) THEN
-        RETURN '';
-    END IF;
-    IF time_value IS NULL THEN
-        RETURN format('%L', NULL);
-    END IF;
-    CASE column_type
-      WHEN 'BIGINT'::regtype, 'INTEGER'::regtype, 'SMALLINT'::regtype THEN
-        RETURN format('%L', time_value); -- scale determined by user.
-      WHEN 'TIMESTAMP'::regtype, 'TIMESTAMPTZ'::regtype THEN
-        -- assume time_value is in microsec
-        RETURN format('%1$L', _timescaledb_internal.to_timestamp(time_value)); -- microseconds
-      WHEN 'DATE'::regtype THEN
-        RETURN format('%L', timezone('UTC',_timescaledb_internal.to_timestamp(time_value))::date);
-      ELSE
-        RETURN time_value;
-    END CASE;
-END
-$BODY$;
+        RETURN QUERY
+        SELECT pg_size_pretty(heap_bytes) as heap_size,
+               pg_size_pretty(index_bytes) as index_size,
+               pg_size_pretty(toast_bytes) as toast_size,
+               pg_size_pretty(total_bytes) as total_size,
+               hd.node_name
+               FROM @extschema@.hypertable_detailed_size(main_table) hd;
 
+END;
+$BODY$;
 
-CREATE OR REPLACE FUNCTION _timescaledb_internal.partitioning_column_to_pretty(
-    d   _timescaledb_catalog.dimension
-)
-    RETURNS TEXT LANGUAGE PLPGSQL STABLE STRICT AS
+CREATE OR REPLACE FUNCTION _timescaledb_internal.chunk_local_size(
+    schema_name_in name,
+    table_name_in name)
+RETURNS TABLE (
+    chunk_id    integer,
+    chunk_name  NAME,
+    heap_bytes bigint,
+    index_bytes bigint,
+    toast_bytes bigint,
+    total_bytes bigint)
+LANGUAGE PLPGSQL STABLE STRICT AS
 $BODY$
-DECLARE
 BEGIN
-        IF d.partitioning_func IS NULL THEN
-           RETURN d.column_name;
-        ELSE
-           RETURN format('%I.%I(%I)', d.partitioning_func_schema, d.partitioning_func, d.column_name);
-        END IF;
-END
+   RETURN QUERY
+   SELECT
+      ch.chunk_id,
+      ch.chunk_name,
+      (ch.total_bytes - COALESCE( ch.index_bytes , 0 ) - COALESCE( ch.toast_bytes, 0 ) + COALESCE( ch.compressed_heap_size , 0 ))::bigint  as heap_bytes,
+      (COALESCE( ch.index_bytes, 0 ) + COALESCE( ch.compressed_index_size , 0) )::bigint as index_bytes,
+      (COALESCE( ch.toast_bytes, 0 ) + COALESCE( ch.compressed_toast_size, 0 ))::bigint as toast_bytes,
+      (ch.total_bytes + COALESCE( ch.compressed_heap_size, 0 ) + COALESCE( ch.compressed_index_size, 0) + COALESCE( ch.compressed_toast_size, 0 ))::bigint as total_bytes 
+   FROM
+   _timescaledb_internal.hypertable_chunk_local_size ch 
+   WHERE
+      ch.schema_name = schema_name_in 
+      AND ch.table_name = table_name_in 
+; 
+END;
 $BODY$;
 
+---should return same information as chunk_local_size--
+CREATE OR REPLACE FUNCTION _timescaledb_internal.chunk_remote_size(
```

Comment:
Same here as above. Why schema+name and not regclass as parameter. Intuitively, I would think this takes a chunk as input, but I don't think this is the case.

Reply:
renamed as chunks_remote_size